### DISCUSSION TITLE
[SOLVED] Handling of /etc/fstab

### DISCUSSION THREAD

**Discussion by FGD-Garuda** [Antynea/grub-btrfs#122] [Q&A] [answered]
Hello!

This tool is really cool, makes my grub menu always updated after my snapshots so I should be able to easily boot from them. I guess that's the idea.

I noticed grub-btrfs does not modify the /etc/fstab file from within the snapshot. This is my fstab for one of my SSD within one snapshot, it still uses the original subvolume references it got snapshoted from:

LABEL=SSD1     /               btrfs   defaults,ssd,noatime,subvol=@           0       1
LABEL=SSD1     /home           btrfs   defaults,ssd,noatime,subvol=@home       0       2
LABEL=SSD1     /.snapshots     btrfs   defaults,ssd,noatime,subvol=@snapshots  0       2
LABEL=SSD1     /var/cache      btrfs   defaults,ssd,noatime,subvol=@varcache   0       2
LABEL=SSD1     /tmp            btrfs   defaults,ssd,noatime,subvol=@tmp        0       2

If I'm booting from a snapshot of "@" (let's say snapshot is called "snapshot_@_20201219") that has a different subvolume name than "@", it has to be addressed here, otherwise "@" will be mounted as root (/) but I need the snapshot to be mounted as root.
Before I try booting from one of the snapshot I want to make sure subvolume "@" will not be mounted to "/", but instead it would be "snapshot_@_20201219".

What should I be expecting?

On the same type of question, how does grub-btrfs know which /home snapshot to mount alongside root?
I have snapshots of both @ and @home, done at the same time. So in /etc/fstab the snapshots names should replace "@" and "@home".

Am I trying to use grub-btrfs for the right purpose?

Thanks!

**Comment by Kr1ss-XD** [comment, marked as answer]
The fstab entry for the `/` mountpoint doesn't have any effect, you could also remove it from the file.
`/` is mounted during the boot process, at a point before the kernel even has access to `fstab` (that file can't be read before the subvolume where it's located is mounted). Which partition/subvolume to mount as `/` is defined in the kernel boot parameters, `grub-btrfs` takes care of this for snapshots. See the entries starting with `linux` inside `grub-btrfs.cfg` in your boot partition.

As for `/home` - since you are doing snapshots anyways at the same time when `/` is snapshotted, maybe consider not having a `@home` subvolume at all, so that your snapshots automatically include the home tree ?

Yes, I guess you _are_ trying to use the script for the right purpose, I'm just not sure if there could be any way how `grub-btrfs` could guess which `/home` snapshot to mount in combination with a certain `/` snapshot, since we can't assume contemporaneous snapshots on every system it runs on (and not even a certain subvolume layout, by the way).

**Comment by FGD-Garuda** [comment]
You're absolutely right about `fstab`, didn't think about that timing situation during boot, it's darn obvious when you think about it! :)

And yes that's also what I thought about `@home`, maybe that's what most `btrfs` people do anyway, I started only a year ago on this file system and was using `ext4` and `xfs` before. Will never go back, though. I guess I tried to replicate the file system structure but in a way it doesn't make sense as I have to maintain 2 snapshots every time with different names and I don't take advantage of whatever advantages there is to segregate both `@` and `@home` since I always keep the pair together, snapshot them together and mount/boot them together. They act as just one big subvolume, actually.

Since having my snapshots in a `grub submenu` has very important uses to me, I will modify my `btrfs` tree and then all will be good. :) Your tool will save me a lot of time and provide me more fun!


Thanks for the very quick reply, man!
Keep up the good work and Happy Holidays!

**Comment by Kr1ss-XD** [comment]
:smile: No, I'm not the author, I'm just a user of this script. Nevertheless I'm glad if this was helpful. Happy holidays to you !
